test/unit/commands_test.rb

```diff
@@ -194,3 +194,87 @@ class TestList < HammerCLIForeman::ListCommand
   end
 
 end
+
+describe HammerCLIForeman::ListCommand do
+  class TestList < HammerCLIForeman::ListCommand
+    resource :domains
+    option '--page', 'PAGE', ''
+    option '--per-page', 'PER_PAGE', ''
+  end
+
+  def build_items(cnt)
+    (1..cnt).map do |i|
+      {:id => i, :name => "Item #{i}"}
+    end
+  end
+
+  def expect_paged_call(page, per_page, item_cnt)
+    api_expects(:domains, :index, "List records page #{page}") do |par|
```

Comment:
I don't have strong preference here but I won't mind leaving it in unit tests as it is tested on abstract command that is not exposed to the user.